EIPS/eip-6963.md

```diff
@@ -158,25 +158,40 @@ interface EIP6963AnnounceProviderEvent extends CustomEvent {
 
 Therefore both wallet providers and the Ethereum library must emit events when they initialize.
 
+Page script listens to the `"eip6963:announceProvider"` event and dispatches the `"eip6963:requestProvider"` event:
 ```typescript
-// Ethereum library initializes
-window.dispatchEvent(new Event("eip6963:requestProvider"));
+// Page script listens to announced providers
+window.addEventListener(
+  "eip6963:announceProvider",
+  (event: EIP6963AnnounceProviderEvent) => {}
+);
 
-// Wallet provider initializes
-window.dispatchEvent(new CustomEvent("eip6963:announceProvider", { detail }));
+// Page script dispatches a request event which will be heard by 
+// content scripts injected earlier
+window.dispatchEvent(new Event("eip6963:requestProvider"));
```

Comment:
I think this should also stay a `CustomEvent` for consistency.

Reply:
Well, the `EIP6963RequestProviderEvent` extends a regular `Event` (which is correct, imo), so I think it should be used here

Reply:
there's no need to use CustomEvent since there's no data being transfered

Reply:
Does it make a difference in terms of interop whether a CustomEvent interface or Event interface is used? Given this is non-normative I think implementers could do either or and it wouldn't impact interoperability here.

Reply:
I like `CustomEvent` better because it better signifies that it's a non-native browser event, plus it's simpler for introspection and TypeScript type narrowing.

Reply:
Also, this would leave open the possibility of dapps exposing metadata about themselves to the wallet providers. You never know 😉

Reply:
@darrylyeo I initially preferred the use of `Event` to follow the rule of least power (i.e. you shouldn't choose `CustomEvent` if a regular `Event` suffices)

But at the same time it does seem like a good idea to pave the road towards dapp being able to send some request data along with the event.

I had an idea of making an additional proposal (or perhaps even updating this one) where a dapp can pass a `locale` in the `eip6963:requestProvider` request, and the wallet would attach a localized `name` field in the `EIP6963ProviderInfo` if it supports that language.

Reply:
Yeah I'd expect nothing to break irregardless of whether `Event` or `CustomEvent` is dispatched from the DApp
Wallets would be expected to check for the `detail` field in case they want to handle it